korman/nodes/node_messages.py

```diff
@@ -877,3 +877,35 @@ def convert_message(self, exporter, so):
         msg.BCastFlags |= (plMessage.kPropagateToModifiers | plMessage.kNetPropagate)
         msg.surface = footstep_surface_ids[self.surface]
         return msg
+
+
+class PlasmaWarpMsgNode(PlasmaMessageNode, bpy.types.Node):
+    bl_category = "MSG"
+    bl_idname = "PlasmaWarpMsgNode"
+    bl_label = "Warp"
+
+    pos_object = PointerProperty(name="Position",
+                                 description="Object defining the target position",
+                                 type=bpy.types.Object)
+
+    output_sockets = OrderedDict([
+        ("receivers", {
+            "text": "Send To",
+            "type": "PlasmaWarpMessageSocket",
+            "valid_link_sockets": {"PlasmaWarpMessageSocket", "PlasmaNodeSocketInputGeneral"},
+        }),
+    ])
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, "pos_object")
+
+    def convert_message(self, exporter, so):
+        msg = plWarpMsg()
+        msg.BCastFlags |= plMessage.kNetPropagate
+        msg.warpFlags |= plWarpMsg.kFlushTransform
+        msg.transform = utils.matrix44(self.pos_object.matrix_local)
```

Comment:
I think this would be more useful if `pos_object` were optional and defaulted to the pose of the object the tree is attached to.

Reply:
That's a useful default.  I'll give it a try and see how that works in practice.